plugins/inputs/wireless/wireless_mac.go

```diff
@@ -0,0 +1,78 @@
+//go:build darwin
+// +build darwin
+
+package wireless
+
+import (
+	"errors"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// default executable path
+const (
+	OSXCMD = "/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport -I"
+)
+
+func (w *Wireless) exe_cmd(cmd string, wg *sync.WaitGroup) ([]byte, error) {
+	parts := strings.Fields(cmd)
+	head := parts[0]
+	parts = parts[1:]
+	out, err := exec.Command(head, parts...).Output()
+	wg.Done() // Need to signal to waitgroup that this goroutine is done
+	return out, err
+}
```

Comment:
I'm not sure why you want to do that since the `Wireless` struct is shared between linux and macOS plugins. having this field in the struct means it would be there on Linux even if it is never used.